### PR TITLE
Add enclosure and escape arguments to importer. Introduce filter to change importer arguments

### PR DESCRIPTION
**Pull request by webspecnick**
This PR is twofold. The first change introduces a new filter to change arguments passed to `WC_Product_CSV_Importer_Controller::get_importer()`. The second change introduces `enclosure` and `escape` arguments to `WC_Product_CSV_Importer` which get passed to `fgetcsv`.

Use case:  
A client is using a legacy inventory tracking system that has trouble outputting text and escaping correctly. We decided that the easiest way to create the import CSV file is to have the legacy system use backticks instead of quotes as the enclosure. I can then use `woocommerce_product_csv_importer_args` to change the enclosure arg to the backtick.

Possible workaround: I could create an additional separate system that converts the CSV from the legacy system into one WooCommerce expects, or I could write my own importer class, perhaps extending the base one and updating `read_file`.

Use case:
A user wants to change the arguments passed to an arbitrary importer class, whether that is a WooCommerce core class or provided by another plugin. The user can use `woocommerce_product_csv_importer_args` to filter and change the arguments passed to it.

Possible workaround: A user can create their own importer class and use `woocommerce_product_csv_importer_class` to change which class is used.

### CODE REVIEW

**Comment by claudiosanches** [comment]
Require fix the coding standards, better read this before: https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md#how-to-submit-a-pr
Let me know when you fix this styling issues.
Thanks for contribute.

File: includes/admin/importers/class-wc-product-csv-importer-controller.php [hunk 1]
```diff
@@ -68,6 +68,7 @@ class WC_Product_CSV_Importer_Controller {
 	 */
 	public static function get_importer( $file, $args = array() ) {
 		$importer_class = apply_filters( 'woocommerce_product_csv_importer_class', 'WC_Product_CSV_Importer' );
+		$args = apply_filters('woocommerce_product_csv_importer_args', $args, $importer_class);
```
Comment (claudiosanches) [inline includes/admin/importers/class-wc-product-csv-importer-controller.php]: Missing spaces after `(` and before `)`.

File: includes/import/class-wc-product-csv-importer.php [hunk 2]
```diff
@@ -39,6 +39,8 @@ public function __construct( $file, $params = array() ) {
 			'update_existing'  => false, // Whether to update existing items.
 			'delimiter'        => ',', // CSV delimiter.
 			'prevent_timeouts' => true, // Check memory and time usage and abort if reaching limit.
+			'enclosure'        => '"', //The enclousure argument passed to fgetcsv.
```
Comment (claudiosanches) [inline includes/import/class-wc-product-csv-importer.php]: Use space after `//` like all the other comments.

File: includes/import/class-wc-product-csv-importer.php [hunk 3]
```diff
@@ -39,6 +39,8 @@ public function __construct( $file, $params = array() ) {
...
+			'escape'           => '\\', //The escape argument passed to fgetcsv.
```
Comment (claudiosanches) [inline includes/import/class-wc-product-csv-importer.php]: The comments here could be more clear than just saying that are passing to `fgetcsv`, or don't need any comment.

**Review by claudiosanches — CHANGES_REQUESTED** [review]

**Comment by webspecnick** [comment]
@claudiosanches I have fixed the styling issues. Thank you for pointing them out.

**Pull request merged**